src/Core/Checkout/Order/SalesChannel/OrderRoute.php

```diff
@@ -113,7 +113,8 @@ public function load(Request $request, SalesChannelContext $context, Criteria $c
 
         $orders = $this->orderRepository->search($criteria, $context->getContext());
 
-        if ($deepLinkFilter !== false) {
+        // remove old orders only if there is a deeplink filter
+        if ($deepLinkFilter === true) {
```

Comment:
The deepLinkFilter will never be true.

Reply:
```suggestion
        if ($deepLinkFilter !== null) {
```

Reply:
It should be a test if its an Instance of or not null 🤦🏻‍♂️